### ISSUE TITLE
"grow run" fails on Windows 10 (in Ubuntu bash)

### ISSUE THREAD

**Issue by pbakaus** [grow/grow#329]
Seems to be related to this error: https://www.bountysource.com/issues/38021250-constructor-for-oserror-used-incorrectly-inotify_c-py-line-406

**Comment by vitorio** [comment]
Are you actually getting an error?

`grow run` launches for me on Windows 10 (native Python and git) but doesn't find any of my content collections, and doesn't report any errors.

What's a good way to debug what grow is doing here?

For reference, my setup is:

 * Install native git via https://git-scm.com/download/win (using git only from the git bash shell)
 * Install native Python 2.7.x via https://www.python.org/downloads/
 * Install Microsoft Visual C++ Compiler for Python 2.7 via https://www.microsoft.com/en-us/download/details.aspx?id=44266 (needed to compile "dulwich")

Then, from the git bash shell:

```console
$ winpty C:\\Python27\\scripts\\easy_install --user pip
$ $APPDATA/Python/Scripts/pip install --user virtualenv
$ cd Documents/
$ mkdir growtest
$ cd growtest
$ $APPDATA/Python/Scripts/virtualenv.exe env
New python executable in C:\Users\vitorio\Documents\growtest\env\Scripts\python.exe
Installing setuptools, pip, wheel...done.
$ ./env/Scripts/pip install grow
Successfully installed Babel-2.2.0 GitPython-1.0.1 GoogleAppEngineCloudStorageClient-1.9.5.0 Jinja2-2.8 Markdown-2.6.6 MarkupSafe-0.19 PyYAML-3.11 SocksiPy-branch-1.1 WebOb-1.3.1 Werkzeug-0.11.10 argh-0.26.2 beautifulsoup4-4.4.0 boltons-16.4.1 boto-2.40.0 certifi-1.0.0 click-4.0 contentful.py-0.9.3 dulwich-0.13.0 ecdsa-0.13 enum34-1.1.1 futures-3.0.5 gcs-oauth2-boto-plugin-grow-1.14 gitdb-0.6.4 google-api-python-client-1.3.1 goslate-1.4.0 grow-0.0.67 html2text-2015.6.21 httplib2-0.9.1 keyring-8.7 keyrings.alt-1.1.1 libsass-0.11.1 oauth2client-2.1.0 paramiko-1.15.1 pathtools-0.1.2 progressbar2-2.6.2 protorpc-0.11.1 pyasn1-0.1.8 pyasn1-modules-0.0.8 pycrypto-2.6.1 pygments-2.0.2 python-dateutil-2.3 pytz-2014.4 pywin32-ctypes-0.0.1 requests-2.3.0 retry-decorator-1.1.0 rsa-3.4.2 semantic-version-2.4.1 six-1.10.0 smmap-0.9.0 texttable-fixed-0.8.3 translitcodec-0.4.0 uritemplate-3.0.0 watchdog-0.7.1 webreview-0.0.63 xtermcolor-1.3
$ git clone #myrepo
Cloning into 'vitor.io'...
Password:
remote: Counting objects: 533, done.
remote: Compressing objects: 100% (527/527), done.
remote: Total 533 (delta 110), reused 0 (delta 0)
Receiving objects: 100% (533/533), 135.86 MiB | 8.69 MiB/s, done.
Resolving deltas: 100% (110/110), done.
Checking out files: 100% (360/360), done.
$ ./env/Scripts/grow run vitor.io/
Warning: libyaml missing, using slower yaml parser.
               Pod: C:\Users\vitorio\Documents\growtest\vitor.io
           Address: http://localhost:8080/index.html
      Server ready. Press ctrl-c to quit.
```
But the web server (whether run from there or from the regular Windows Command Prompt) reports:

> Internal Server Error
> The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.

And http://localhost:8080/_grow only shows my four static routes, and no content collections.

**Comment by pbakaus** [comment]
@vitorio the first difference in my setup is that I'm not using the Git Bash, but the integrated Ubuntu bash (see https://msdn.microsoft.com/en-us/commandline/wsl/install_guide).

Luckily, I'm happy to report that in the latest Windows 10 Insider builds, the error I reported is fixed, and grow runs fine, without any special setup. I encourage you to give the Ubuntu bash a try!
